### DISCUSSION TITLE
Way to define cells as "row header"s (th scope="row")?

### DISCUSSION THREAD

**Discussion by Josh68** [TanStack/table#3287] [Q&A]
Hello. I'm new to RT and feel like this must be a stupid question, but I've tried searching the docs, API docs, and this discord, and I'm coming up empty. Is there a prescriptive way to define row cells (usually those in the first column) in even the most basic table, as "row headers" (i.e., `th`  with `scope="row"` rather than as plain `td`s? I'm currently doing this just by passing a prop like `rowHeaders` to the main component I'm using to render RT data, but somehow it seems to me like it should be something defined in my RT columns array, e.g. `{accessor: unique_key, Header: string, rowHeader: boolean}`. Any advice, or is the way I'm already doing it basically the way it's done? Thanks

**Comment by jrmyio** [comment]
Also just came accross this, considering its an actual aria role it should really be part of react-table.

More on it here:
https://www.w3.org/TR/wai-aria-1.1/#rowheader

**Comment by qmotas** [comment]
Is there any additional information or progress on this discussion?

For reference, React Aria's `<Column>` component has the `isRowHeader` property that might be relevant to this discussion.
https://react-spectrum.adobe.com/react-aria/Table.html#column

**Reply by qmotas** [reply]
My workaround here:
https://github.com/qmotas/tanstack-table-rowheader-poc

```ts
declare module "@tanstack/react-table" {
  interface ColumnMeta<TData, TValue> {
    isRowHeader?: boolean;
  }
}
```

```ts
columnHelper.accessor("name", {
  header: () => "Name",
  cell: (info) => info.getValue(),
  meta: {
    isRowHeader: true,
  },
}),
```

```tsx
<td
  key={cell.id}
  role={cell.column.columnDef.meta?.isRowHeader
    ? "rowheader"
    : undefined}
>
```

extending `ColumnMeta` with `isRowHeader` property

**Reply by Josh68** [reply]
Thanks. This sounds like a more formal (or tidy) way of approaching the problem, otherwise similar to what I was already doing. We're still on v7, so I don't think this API actually applies in that case. Either way, I still feel like this could be simpler.
